### ISSUE TITLE
HEEx doesn't raise  No space between attributes error

### ISSUE THREAD

**Issue by bcardarella** [phoenixframework/phoenix_live_view#3528]
This works just fine:

```heex
~H"""
   <div style=""id="123"/>
"""
```

It will render as :

```html
<div style="" id="123"/>
```

but the original template is invalid and shouldn't compile.

**Comment by josevalim** [comment]
Although not ambiguous here, I understand why HTML does it. We could error but perhaps it makes more sense to warn, given the formatter will automatically fix it for you anyway, and that would not be possible if erroring.

**Comment by bcardarella** [comment]
Now that LV is 1.0 I agree that this should not error. Considering it compiles anyway I didn't know if it was worth pursuing or not but figured I'd bring attention to it.

**Comment by bcardarella** [comment]
And Floki parses this syntax just fine withour warning, which means that Mochi parses just fine. I think I'll close this as a non-issue.

I will have to account for this in LVN's template parser as we are erroring on it.

**Comment by josevalim** [comment]
Let's keep it open because I think warning is better. We don't want to have someone thinking this is fine and then emitting some hand written HTML somehow with this bug :)

**Comment by josevalim** [comment]
FWIW, browsers parse it as multiple attributes too, but they are also historically known to be very lenient.

**Comment by bcardarella** [comment]
yeah I sent it through the W3C validator which will be less forgiving

**Comment by Schultzer** [comment]
This looks like it could be non trivial to implement.

There is a couple of rules:

1. attributes value require to end on ```" ```.
2. attributes value require to end on ```"\```.
3. attributes value require to end on ```">```.
4. attributes values ending on a..z should errors.
